UnityProject/Assets/Scripts/Player/MovementV2/MovementSynchronisation.cs

```diff
@@ -1353,13 +1353,14 @@ public bool CanInPutMove(bool queueing = false)
 		return false;
 	}
 
+	[Command]
```

Comment:
why is this a command, all you got to do is return a move request  when you try walking into electric grill

Reply:
> this has a bug where it doesn't trigger on server because the client doesn't send the move request

???

Reply:
is this what you want now?

Reply:
```
			var count = Bumps.Count;
				for (int i = count - 1; i >= 0; i--)
				{
					if (i >= Bumps.Count) continue;

					Bumps[i].OnBump(this.gameObject, byClient);
					bumpedSomething = true;
				}
				```
So, there is already code for handling bumps, you just need integrated as part of the tile move,

IsPassableAtAllMatricesV2 
need to pass in the bumps 
to matrixTarget.Matrix.MetaTileMap.IsPassableAtOneTileMapV2

 then add your custom cheque in IsPassableAtOrthogonalTileV2 , that would add to bumps